backend/src/main/java/com/zeros/backend/services/impl/SimpleRestaurantAffiliateNetwork.java

```diff
@@ -0,0 +1,50 @@
+package com.zeros.backend.services.impl;
+
+import com.sun.org.apache.regexp.internal.RE;
+import com.zeros.backend.exceptions.AlreadyAffiliatedRestaurantException;
+import com.zeros.backend.exceptions.NullQrException;
+import com.zeros.backend.exceptions.RestaurantNotFoundException;
+import com.zeros.backend.models.RestaurantAffiliate;
+import com.zeros.backend.repositories.RestaurantAffiliateRepository;
+import com.zeros.backend.repositories.RestaurantRepository;
+import com.zeros.backend.services.RestaurantAffiliateNetworwk;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+
+@Service
+public class SimpleRestaurantAffiliateNetwork implements RestaurantAffiliateNetworwk {
+
+    RestaurantRepository restaurantRepository;
+    RestaurantAffiliateRepository restaurantAffiliateRepository;
+
+    @Autowired
+    public SimpleRestaurantAffiliateNetwork(RestaurantRepository restaurantRepository,RestaurantAffiliateRepository restaurantAffiliateRepository) {
+        this.restaurantRepository = restaurantRepository;
+        this.restaurantAffiliateRepository = restaurantAffiliateRepository;
+    }
+
+    @Override
+    public RestaurantAffiliate affiliate(Long restaurantId, Long qrCode) throws RestaurantNotFoundException {
+
+        if (qrCode == null) throw new NullQrException("Qr code must be present");
+
+        boolean restaurantNotExists = !restaurantRepository.exists(restaurantId);
```

Comment:
I think that there is no need to declare a boolean for this condition, what do you think @JumperSpirit  ?

Reply:
No